source/features/download-folder-button.tsx

```diff
@@ -24,14 +26,16 @@ function add(folderDropdown: HTMLElement): void {
 
 function init(signal: AbortSignal): void {
 	observe([
-		'[title="More options"]',
+		'[title="More options"]',	// PageDetect.isRepoTree
+		'[title="More file actions"]', // PageDetect.isSingleFile
 		'[aria-label="Add file"] + details', // TODO: Drop in mid 2023. Old file view #6154
 	], add, {signal});
 }
 
 void features.add(import.meta.url, {
 	include: [
 		pageDetect.isRepoTree,
+		pageDetect.isSingleFile,
```

Comment:
This is a hack, a feature should not be enabled on random pages just to cleanup after itself

Reply:
Yes, but add singleFile is needed.
When we access blob direct link, Download button will not be rendered.

Reply:
This feature must appear exclusively on directories because it _downloads directories_ and the `include` array tells me exactly that.

It's only "needed" here because the feature manager is broken, but it's still a hack that we should not use